articles/openshift/howto-add-or-update-your-pull-secret.md

```diff
@@ -0,0 +1,203 @@
+
+```json
+{
+    "auths": {
+        "cloud.openshift.com": {
+            "auth": "<my-crc-secret>",
+            "email": "klamenzo@redhat.com"
+        },
+        "quay.io": {
+            "auth": "<my-quayio-secret>",
+            "email": "klamenzo@redhat.com"
+        },
+        "registry.connect.redhat.com": {
+            "auth": "<my-registry.connect.redhat.com-secret>",
+            "email": "klamenzo@redhat.com"
+        },
+        "registry.redhat.io": {
+            "auth": "<my-registry.redhat.io-secret>",
+            "email": "klamenzo@redhat.com"
+        }
+    }
+}
+```
+
+3. Edit the pull secret file you got from your cluster by adding in the entries found in your Red Hat pull secret. Your final file should look like the following (note that the actual secret values have been removed):
+
+```json
+{
+    "auths": {
+        "cloud.openshift.com": {
+            "auth": "<my-crc-secret>",
+            "email": "klamenzo@redhat.com"
+        },
+        "quay.io": {
+            "auth": "<my-quayio-secret>",
+            "email": "klamenzo@redhat.com"
+        },
+        "registry.connect.redhat.com": {
+            "auth": "<my-registry.connect.redhat.com-secret>",
+            "email": "klamenzo@redhat.com"
+        },
+        "registry.redhat.io": {
+            "auth": "<my-registry.redhat.io-secret>",
+            "email": "klamenzo@redhat.com"
+        },
+        "arosvc.azurecr.io": {
+            "auth": "<my-aroscv.azurecr.io-secret>"
+        }
+    }
+}
+```
+
+4. Ensure the the file is valid json. There are many ways to validate your json. The following example uses jq:
+```json
+cat pull-secret.json | jq
+```
+
+> NOTE
+> If an error is in the file it can be seen `parse error`.
+
+## Add your pull secret to your cluster
+
+Run the following update command to update or add your pull secret to your cluster:
+
+```console
+oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=./pull-secret.json
+```
+
+Once the secret is set you are ready to enable Red Hat certified operators.
+
+### Modify the configuration files
+
+Modify the following objects to enable Red Hat Operators.
+
+First, modify the Samples Operator configuration file. Run the following command to edit the configuration file:
+
+```
+oc edit configs.samples.operator.openshift.io/cluster -o yaml
+```
+
+Change the `spec.architectures.managementState` and the `status.architecture.managementState` values from `Removed` to `Managed`. 
+
+The following shows only the relevant sections of the edited yaml file.
+
+```yaml
+apiVersion: samples.operator.openshift.io/v1
+kind: Config
+metadata:
+  
+  ...
+  
+spec:
+  architectures:
+  - x86_64
+  managementState: Managed
+status:
+  architectures:
+
+  ...
+
+  managementState: Managed
+  version: 4.3.27
+```
+
+Save the file to apply your edits.
```

Comment:
this could be expressed in a single `oc patch` command - might be simpler ...

Reply:
Let's do this in a follow-up PR. I have some questions on how to make it work:
- Do I need 2 patch commands or can I update two separate places in one YAML file at once. (Does the customer even need to update status.managementState?)
- What do I need to add to make this command work?
```
[klamenzo@klamenzo aro]$ oc patch configs.samples.operator.openshift.io/cluster -p '{"spec":{"managementState":"Managed"}}'
Error from server (UnsupportedMediaType): the body of the request was in an unknown format - accepted media types include: application/json-patch+json, application/merge-patch+json, application/apply-patch+yaml
```